tests/test_parse.py

```diff
@@ -0,0 +1,167 @@
+import pytest
+
+from boolean import BooleanAlgebra
+from boolean import ParseError
+
+class TestParse:
+    def test_constant(self):
+        algebra = BooleanAlgebra()
+
+        for x in ['0', 'false']:
+            assert algebra.parse(x) == algebra.FALSE
+            assert algebra.parse('(' + x + ')') == algebra.FALSE
+            assert algebra.parse('((' + x + '))') == algebra.FALSE
+            assert algebra.parse('(((' + x + ')))') == algebra.FALSE
+
+        for x in ['1', 'true']:
+            assert algebra.parse(x) == algebra.TRUE
+            assert algebra.parse('(' + x + ')') == algebra.TRUE
+            assert algebra.parse('((' + x + '))') == algebra.TRUE
+            assert algebra.parse('(((' + x + ')))') == algebra.TRUE
+
+    @pytest.mark.xfail(reason='IndexError indicates parsing problem')
```

Comment:
If you remove the `xfail` decorator, the parser breaks and the whole thing crashes with an `IndexError`. That is not supposed to happen because the parser was designed to throw custom errors when there is a problem. So, you either throw your own errors or generic errors, not both (imho). Since it starts to throw generic errors all over sudden means there is likely some corner case that was not considered.